app/Http/Controllers/UserController.php

```diff
@@ -65,15 +65,60 @@ public function showCategories()
     public function makeLesson($categoryId)
     {
         $user = auth()->user();
-    
+        
         $lesson =Lesson::create([
-            'score' = > 0,
-            'category_id' = > $categoryId,
-            'user_id' = > $user->id,
+            'category_id' => $categoryId,
+            'user_id' => $user->id,
         ]);
-       
+
         $lessonId=$lesson->id;
 
         return redirect()->action('UserController@showQuiz',compact('categoryId','lessonId'));
     }
+
+    public function showQuiz($categoryId,$lessonId)
+    {
+        $currentUserId = auth()->user()->id;
+
+        $categoryId=(int)$categoryId;
+
+        $questions = Question::where('category_id', $categoryId)->paginate(1);
+        $currentPage = $questions->currentPage();
+        
+        return view('users.Quiz',compact('questions','categoryId','lessonId','currentPage'));
+    }
+
+    public function store(Request $request)
+    {
+        $user = auth()->user();
+        $categoryId = $request->categoryId;
```

Comment:
I suggest that a function must only do one thing. In this code it is doing a lot of things. Think of a way that you can express the code in a single job